include/aws/io/io.h

```diff
@@ -134,6 +134,7 @@ enum aws_io_errors {
     AWS_IO_TLS_ALERT_NOT_GRACEFUL,
     AWS_IO_MAX_RETRIES_EXCEEDED,
     AWS_IO_RETRY_PERMISSION_DENIED,
+    AWS_IO_PKCS11_ERROR,
```

Comment:
Do you see a distinction here between user and service errors? Like CKR_BUFFER_TOO_SMALL is our problem and CKR_PIN_INCORRECT is users problem.

Reply:
Yeah, this is the part of the PR I'm least confident in, and totally open to changing it going forward.

Each of our C libraries is limited to 1024 unique error codes. Adding every single PKCS#11 error code like: `CKR_UNWRAPPING_KEY_TYPE_INCONSISTENT -> AWS_IO_PKCS11_ERROR_UNWRAPPING_KEY_TYPE_INCONSISTENT`
would take up about 100 slots, so it's possible.

Hopefully issues like CKR_BUFFER_TOO_SMALL won't surface because we'll make the calls correctly. But anything can happen because each vendor has their own implementation.